cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -116,18 +115,18 @@ public CruiseControlApiImpl(Vertx vertx) {
                         response.bodyHandler(buffer -> {
                             String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
                             JsonObject json = buffer.toJsonObject();
-                            CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
                             if (json.containsKey(CC_REST_API_ERROR_KEY)) {
                                 // If there was a client side error, check whether it was due to not enough data being available
                                 if (json.getString(CC_REST_API_ERROR_KEY).contains("NotEnoughValidWindowsException")) {
+                                    CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
                                     ccResponse.setNotEnoughDataForProposal(true);
                                     result.complete(ccResponse);
                                 } else {
                                     // If there was any other kind of error propagate this to the operator
                                     result.fail(json.getString(CC_REST_API_ERROR_KEY));
                                 }
                             } else {
-                                result.complete(ccResponse);
+                                result.fail(json.toString());
```

Comment:
Probably worth having an exception class for this, since `fail(String)` uses the unhelpful `NoStackTraceThrowable` which (I'm guessing) could propagate to the CR status condition.

Reply:
Good point. Will use the (already created) `CruiseControlRestException` that I should have used in the first place!